src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets

```diff
@@ -258,6 +258,9 @@
       <Delete Files="@(SqlArtifacts -> '$(TargetDir)%(Filename)%(Extension)')" />
   </Target>
 
+  <!-- Allows a project to set to publish on build for their sql project (so they can deploy it on build that is if they want). -->
+  <Target Name="PublishSqlProject" BeforeTargets="CoreCompile" DependsOnTargets="Publish" Condition="'$(PublishOnBuild)' != 'False' And '$(DeployOnPublish)' != 'False'" />
```

Comment:
Looking at the `Condition` here I get the feeling that it is an opt-out now, rather than an opt-in. To be honest I think this should be an opt-in.

Additionally it seems we have two separate properties that achieve the same thing?

Reply:
I could do it like that however it seems the Publish target runs when ``DeployOnPublish`` is ``true``, while this might try to run the target it might not run if the user sets ``DeployOnPublish`` to ``false``.

Reply:
Removed ``DeployOnPublish`` from this line. I wonder if it should also be removed from the publish target and instead checked within it where it will try to deploy it if it's not set.